backend/FwLite/FwDataMiniLcmBridge/Api/FwDataMiniLcmApi.cs

```diff
@@ -104,19 +104,23 @@ public Task<WritingSystems> GetWritingSystems()
             .Select(ws => ws.Id).ToHashSet();
         var writingSystems = new WritingSystems
         {
-            Vernacular = WritingSystemContainer.CurrentVernacularWritingSystems.Select(FromLcmWritingSystem).ToArray(),
-            Analysis = Cache.ServiceLocator.WritingSystems.CurrentAnalysisWritingSystems.Select(FromLcmWritingSystem).ToArray()
+            Vernacular = WritingSystemContainer.CurrentVernacularWritingSystems.Select(definition =>
+                FromLcmWritingSystem(definition, WritingSystemType.Vernacular)).ToArray(),
+            Analysis = Cache.ServiceLocator.WritingSystems.CurrentAnalysisWritingSystems.Select(definition =>
+                FromLcmWritingSystem(definition, WritingSystemType.Analysis)).ToArray()
         };
         CompleteExemplars(writingSystems);
         return Task.FromResult(writingSystems);
     }
 
-    private WritingSystem FromLcmWritingSystem(CoreWritingSystemDefinition ws)
+    private WritingSystem FromLcmWritingSystem(CoreWritingSystemDefinition ws, WritingSystemType type)
     {
         return new WritingSystem
         {
+            Id = Guid.Empty,
+            Type = type,
             //todo determine current and create a property for that.
```

Comment:
ahh, in FW in the WS list you can check or uncheck the writing systems, this is represented in code by the WS being in a list called Current writing systems. Rather than doing that I would prefer to have a property to track that, it's just not implemented